flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,204 @@
+
+                        logger.verbose('Basic SSL/TLS encrypted communications enabled')
+                        logger.verbose('TLS insecure cert mode enabled. Broker cert will not be validated')
+
+                    #Handle SSL/TLS communication with certificate authentication
+                    if self.config.get('certificates',False):
+                        certs = self.config.get('certificates',{})
+                        logger.debug('TLS certificate config: {}'.format(str(certs)))
+
+                        tls_version_map = {'tlsv1.2': ssl.PROTOCOL_TLSv1_2, 'tlsv1.1': ssl.PROTOCOL_TLSv1_1, 'tlsv1': ssl.PROTOCOL_TLSv1, '': None}
+                        tls_version = tls_version_map.get(certs.get('tls_version'),ssl.PROTOCOL_TLSv1_2)
+                        logger.verbose('TLS version is {}'.format(str(tls_version)))
+
+                        if certs.get('validate_broker_cert',True):
+                            cert_required = ssl.CERT_REQUIRED
+                        else:
+                            cert_required = ssl.CERT_NONE
+                        logger.verbose('TLS cert_required={}'.format(str(cert_required)))
+
+                        self.tls_set(ca_certs=certs.get('broker_ca_cert',None),
+                                      certfile=certs.get('client_cert',None),
+                                      keyfile=certs.get('client_key',None),
+                                      cert_reqs=cert_required,
+                                      tls_version=tls_version)
+
+                        if not certs.get('validate_broker_cert',None):
+                            self.tls_insecure_set(True)
+
+                            logger.debug('TLS insecure cert mode enabled. Broker cert will not be validated')
+                        else:
+                            logger.debug('TLS secure cert mode enabled. Broker cert will be validated')
+
+                    #Handle user/pass authentication
+                    if self.config.get('username',False) or self.config.get('password',False):
+                        logger.verbose('Credential passwords s are redacted to protect the innocent...')
+                        logger.verbose('Auth credentials: username=[{}] password sha256 hash is "{}"'.format(self.config.get('username'),sha256(str(self.config.get('password','')).encode('utf-8')).hexdigest()))
+                        logger.verbose('You can validate them yourself by calculating the sha256 hex digest of your password string (google is your friend if you do not know how to do this)')
+                        logger.verbose('Note: a password that is not provided (i.e. blank) will hash to "{}"'.format(sha256(str('').encode('utf-8')).hexdigest()))
+
+                        self.username_pw_set=(self.config.get('username',None),self.config.get('password',None))
+
+                    logger.verbose("Connecting to {}:{}".format(self.config.get('broker_address'),str(self.config.get('broker_port'))))
```

Comment:
this is relevant to other places as well: no need to use `.format()` for logger messages, just use curly brackets and pass the args to the logger itself. https://github.com/Delgan/loguru#modern-string-formatting-using-braces-style

Reply:
done with commit 1ae4cbe5777ff1c8bdc2ff5c915f8701c3b5d191